curves/bn254/Cargo.toml

```diff
@@ -16,16 +16,19 @@ edition = "2021"
 ark-ff = { version= "0.4.0", default-features = false }
 ark-ec = { version= "0.4.0", default-features = false }
 ark-std = { version = "0.4.0", default-features = false }
+ark-r1cs-std = { version = "0.4.0", default-features = false, optional = true }
 
 [dev-dependencies]
 ark-serialize = { version = "0.4.0", default-features = false }
 ark-algebra-test-templates = { version = "0.4.0", default-features = false }
 ark-algebra-bench-templates = { version = "0.4.0", default-features = false }
+ark-curve-constraint-tests = { path = "../curve-constraint-tests", default-features = false }
+ark-relations = { version = "0.4.0", default-features = false }
 
 [features]
 default = [ "curve" ]
 std = [ "ark-std/std", "ark-ff/std", "ark-ec/std" ]
```

Comment:
consider adding `ark-r1cs-std?/std` in case `std` feature is enabled

Reply:
Notice that if we don't require r1cs feature. Then loading `std` should not load this no?

I think it makes sense as it is. 
`r1cs` enables `ark-r1cs-std` and `std` enables `[ "ark-std/std", "ark-ff/std", "ark-ec/std" ]` (no r1cs unless it is explicitly required).

Reply:
The `?` will activate `std` for `ark-r1cs-std` only if `r1cs` is enabled

Reply:
Ohhh! I didn't know that `?` thing!! Fixed in 504a053